lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java

```diff
@@ -64,11 +71,47 @@ public static boolean isValidURI(String uriString) {
    * @param path the path as a String
    * @return the correct InputStream
    */
-  public static InputStream getInputStream(String path) throws IOException {
+  public static InputStream getInputStream(String path, FileSystemManager fsManager) {
+    InputStream is;
+
+    // setting up VFS
     if (isValidURI(path)) {
-      return VFSInputStream.open(path);
+      FileObject file;
+      try {
+        file = fsManager.resolveFile(path);
+
+        if (file == null) {
+          log.warn("File object is null for path: {}", path);
+          return null;
+        }
+
+        if (!file.exists()) {
+          log.warn("File not found at: {}", path);
+          return null;
+        }
+
+        FileContent content = file.getContent();
+        try {
+          is = content.getInputStream();
+          return is;
+        } catch (FileSystemException e) {
```

Comment:
consider removing this inner try/catch since the outer try/catch is for the same exception type